keras/callbacks.py

```diff
@@ -2767,6 +2767,18 @@ def on_train_batch_end(self, batch, logs=None):
                 1.0 / batch_run_time,
                 step=self._train_step,
             )
+
+        should_record = False
+        if type(self.update_freq) == int:
+            should_record = tf.equal(self._train_step % self.update_freq, 0)
+        with tf.summary.record_if(should_record):
+            if logs:
+                with self._train_writer.as_default():
+                    for name, value in logs.items():
```

Comment:
Thanks for the PR! Unfortunately this would have a performance hit with asynchronous strategies such as `TPUStrategy` and `ParameterServerStrategy`, and users would need to add the summary logging ops within custom `train_step()` or `call()`. @nkovela1 is actively working on an example workflow users could use in the documentation.

On the other hand, feel free to do this in a custom callback inheriting `TensorBoard`, if your workflow doesn't involve asynchronous tf.distribute strategies.